### PR TITLE
Mat 5350 set measure text in bundle 2

### PR DESCRIPTION
**Pull request by sb-cecilialiu**
## MADiE FHIR SERVICE

Jira Ticket: [MAT-5350](https://jira.cms.gov/browse/MAT-5350)
(Optional) Related Tickets:

### All Submissions
* [x] This PR has the JIRA linked.
* [x] Required tests are included.
* [x] No extemporaneous files are included (i.e Complied files or testing results).
* [x] This PR is merging into the **correct branch**.
* [ ] All Documentation needed for this PR is Complete (or noted in a TODO or other Ticket).
* [ ] Any breaking changes or failing automations are noted by placing a comment on this PR.

### DevSecOps
If there is a question if this PR has a security or infrastructure impact, please contact the Security or DevOps engineer assigned to this project to discuss it further.

* [x] This PR has NO significant security impact (i.e Changing auth methods, Adding a new user type, Adding a required but vulnerable package).

### Reviewers
By Approving this PR you are attesting to the following:

*  Code is maintainable and reusable, reuses existing code and infrastructure where appropriate, and accomplishes the task’s purpose.
*  The tests appropriately test the new code, including edge cases.
*  If you have any concerns they are brought up either to the developer assigned, security engineer, or leads.

### CODE REVIEW

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java [hunk 1]
```diff
@@ -135,4 +144,40 @@ void addBytesToZip(String path, byte[] input, ZipOutputStream zipOutputStream)
   private String convertFhirResourceToString(Resource resource, IParser parser) {
     return parser.setPrettyPrint(true).encodeResourceToString(resource);
   }
+
+  protected DomainResource setMeasureTextInBundle(
+      Bundle bundle, String measureId, String humanReadableStr) {
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: does measureId not available in a measure resource in a bundle?

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java [hunk 2]
```diff
@@ -135,4 +144,40 @@ void addBytesToZip(String path, byte[] input, ZipOutputStream zipOutputStream)
...
+
+    Optional<Bundle.BundleEntryComponent> measureEntryOpt =
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java:157]: I believe `getMeasureEntry` method is already present in one of the services(maybe hr). can reuse it.

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java [hunk 3]
```diff
@@ -135,4 +144,40 @@ void addBytesToZip(String path, byte[] input, ZipOutputStream zipOutputStream)
...
+    return null;
+  }
+
+  protected Narrative createNarrative(String id, String humanReadableStr) {
+
+    try {
+      Document doc = Jsoup.parse(humanReadableStr);
+      doc.outputSettings().syntax(Document.OutputSettings.Syntax.xml);
+      doc.outputSettings().escapeMode(org.jsoup.nodes.Entities.EscapeMode.xhtml);
+      String divContent = "<div>" + doc.select("body").html() + "</div>";
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: not sure this is needed if you use `narrative.setDivAsString`. can you not pass the whole hr string?

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 1 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: @adongare No it's not in the measure resource, this is the madie measure id, used for logging exceptions.

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 2 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java:157]: @adongare You are right, I changed it to reuse the method from HumanReadableService, thanks Ashok!

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 3 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: @adongare No this is needed, and if I just pass the whole hr string, I would get this error: java.lang.RuntimeException: org.hl7.fhir.exceptions.FHIRFormatError: Unable to Parse HTML - starts with 'null::html' not 'div' at line 2 column 7
	at org.hl7.fhir.utilities.xhtml.XhtmlNode.setValueAsString(XhtmlNode.java:476) ~[org.hl7.fhir.utilities-5.6.97.jar:na]
	at org.hl7.fhir.r4.model.BaseNarrative.setDivAsString(BaseNarrative.java:50) ~[org.hl7.fhir.r4-5.6.97.jar:na]

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 1 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: yeah, you are right. My bad.

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 3 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: I was wondering if the wrapper template that we added for css is causing the issue. We'll have to confirm with Stan once if the HR with CSS needs to be set as narrative or just the o/p of liquid template.

Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: checked a few examples. they are without CSS. setting o/p from just liquid template http://hl7.org/fhir/us/cqfmeasures/Measure-BreastCancerScreeningFHIR.json.html.

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: @adongare I just made the change to separate human readable string from human readable string with css style, based on your suggestion, thanks so much Ashok!

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java [hunk 4]
```diff
@@ -135,4 +146,34 @@ void addBytesToZip(String path, byte[] input, ZipOutputStream zipOutputStream)
...
+  protected DomainResource setMeasureTextInBundle(
+      Bundle bundle, String measureId, String humanReadableStr) {
+
+    Optional<Bundle.BundleEntryComponent> measureEntryOpt =
+        humanReadableService.getMeasureEntry(bundle);
+
+    if (measureEntryOpt.isPresent()) {
+      DomainResource dr = (DomainResource) measureEntryOpt.get().getResource();
+      dr.setText(createNarrative(measureId, humanReadableStr));
+      dr.getText().setStatus(NarrativeStatus.GENERATED);
```
Comment (adongare) [inline src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java:163]: seems like a duplicate. you are already setting it in `createNarrative`

Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java:163]: @adongare The Narrative is set with hr, but we need to set the Narrative to measure resource setText, that's what MAT-5350 requires right?

Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java:163]: that is right. I'm talking about this line `dr.getText().setStatus(NarrativeStatus.GENERATED);` you are basically getting the narrative and setting the status to generated. My understanding is you already have narrative status(line 163). this is just overriding it, right?

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 3 above)
Comment (adongare) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java]: np, thank you!

File: src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java (hunk 4 above)
Comment (sb-cecilialiu) [reply src/main/java/gov/cms/madie/madiefhirservice/services/ExportService.java:163]: @adongare Ok thanks, I misunderstood it. I just removed the overriding part. Thanks Ashok!

**Review by adongare — APPROVED** [review]
looks good to me

**Pull request merged**